js/map.mjs

```diff
@@ -49,22 +52,20 @@ function updateRandomGateCount(Found){
     }
 
     if(randomGateCount >= 40){
-        document.getElementById("random_Gate_Count").innerText = randomGateCount + "✔";
+        document.getElementById("random_Gate_Count").innerText = getRandomGateCount() + "✔";
         document.getElementById("random_Gate_Count").style = "color:green";
     }
     else{
-        document.getElementById("random_Gate_Count").innerText = randomGateCount;    
+        document.getElementById("random_Gate_Count").innerText = getRandomGateCount();    
         document.getElementById("random_Gate_Count").style = "color:black";
     }
 }
 function initRandomGateCount(){
     //Init randomGateCount. 
-        //Is there a better way to do this? Only issue I think this presents is if we change the location IDs.
-    for(let i = 189; i <= 268; i++){
-        if(window.savedata.location[i]){
-            randomGateCount++;
-        }
-    }
+    let root = window.findOnTree(window.jsondata.location, x=>x.name == "Random Gates", y=>y.name == "Random Gates" || y.elements == null)
+    let completed = window.sumCompletionItems(root);
+    randomGateCount = completed[0];
```

Comment:
also did you check that this actually works? I wrote it off the top of my head without testing

Reply:
It works. I doubled checked with a few tests/console.logs as well.